### DISCUSSION TITLE
how do I get the UIImage for existing KFImage

### DISCUSSION THREAD

**Discussion by neal3000** [onevcat/Kingfisher#2115] [Q&A]
I have a KFImage passed to a child view
from the child view I want to access the underlying UIImage , so I can use it with SharkLink api

But I cannot figure out how.

Thanks for any help

**Comment by onevcat** [comment]
Not 100% sure if this meets your need, but how about:

```swift
struct ContentView: View {
    @State private var loadedImage: UIImage?

    var body: some View {
        KFImage(.init(string: "https://onevcat.com/assets/images/avatar.jpg"))
            .onSuccess {
                loadedImage = $0.image
            }
        if let loadedImage {
            ChildView(image: .init(uiImage: loadedImage))
        }
    }
}

struct ChildView: View {
    let image: Image

    var body: some View {
        ShareLink("Share Me", item: image, preview: SharePreview("Preview", image: image))
    }
}
```

**Reply by neal3000** [reply]
Thanks, but what I'm trying to do is avoid passing the `UIImage` (the hierarchy is deep and I'm dealing with an array of KFImage at the Parent)

Ideally I'd like

```swift
Parent : View {
   var images : \[ KFImage \]()
 var body: some View {
  ..
 NavigationLink {  ImageDetailView(images, index) }
}
 
ImageDetailView   : View {
   @State var images : [KFImage]
  @State  var index : Int 
 var body: some View {
    ...
   let uuimage = images[index].uiimage                 <-----  The is the part I don't know how to do
 ShareLink( "",item: uuimage)
```

**Reply by onevcat** [reply]
It is not the way how SwiftUI works. The `KFImage`, as a `SwiftUI.View`, does not hold the underneath image object. That is similar that you cannot get the text from a `Text` or get a boolean from a `Toggle`. Using the `KFImage` in that way (and passing it from a view to another) is not preferred and anti-pattern.

**Reply by neal3000** [reply]
Thanks, guess I'll have to refactor
